docs/caching.md

```diff
@@ -0,0 +1,51 @@
+# HTTP requests caching
+
+## Behavior
+
+We currently use Nginx to cache HTTP requests. It works like a standard reverse proxy: when a request comes in, a cache key is generated from various information sent by the client, a cached response is returned if one is found, otherwise the request is forwarded to the origin and returned to the client while being put in cache if appropriate.
+
+For the cache key, we take into account the following parameters:
+
+- Country corresponding to the client IP using GeoIP
+- Value of the `frontend_active_experiments` cookie (parsed from the `Cookie` request header) - its absence being a separate value as far as the cache key is concerned
```

Comment:
Would be good to cover why we don't need to vary on Cookie and that relationship with caching headers so 3rd party proxies aren't going to cache incorrect responses for related to different cookie values.

For me the parts that are hardest to reason about are the differences between what we doing internally and the parts that are visible to clients.

Reply:
A true `Vary` on `Cookie` essentially makes the cache key different for everybody, since it would depend on all cookies sent to AMO, including the Google Analytics cookie.

Reply:
Yep, and I remember you mentioning how caches outside of our control won't cache (e.g. 3rd party proxies) because of the headers sent, so that would prevent a case where without the vary on Cookie a caching proxy could cache an incorrect response for different Cookies sent in the request. Is that something that could be detailed here.

Reply:
https://github.com/mozilla/addons-frontend/pull/10777/commits/085c64bcddb19e749035662f155cc0fb927be4a7 and https://github.com/mozilla/addons-frontend/pull/10777/commits/89d316c998a8c2d091516a86443f42f890337659 should explain that